src/vumi2/applications/junebug_message_api/junebug_message_api.py

```diff
@@ -0,0 +1,163 @@
+    # mo_message_url_timeout: int = 10
+    # # Maximum time (seconds) a mo_message_url is allowed to take to process an event
+    # event_url_timeout: int = 10
+
+    def url(self, path: str) -> str:
+        return f"{self.base_url_path}{path}"
+
+
+class JunebugMessageApi(BaseWorker):
+    """
+    An implementation of the Junebug HTTP message API.
+
+    Inbound messages and events are sent over HTTP to the configured
+    URL(s). Outbound messages are received over HTTP and sent to the
+    configured transport.
+
+    TODO: Finish implementation.
+    """
+
+    config: JunebugMessageApiConfig
+
+    async def setup(self) -> None:
+        state_cache_class = class_from_string(self.config.state_cache_class)
+        self.state_cache = state_cache_class(self.config.state_cache_config)
+        self.connector = await self.setup_receive_inbound_connector(
+            self.config.connector_name,
+            self.handle_inbound_message,
+            self.handle_event,
+        )
+        # sm_url = self.config.url("/messages")
+        # self.http_app.add_url_rule(sm_url, view_func=self.http_send_message)
+
+    async def handle_inbound_message(self, message: Message) -> None:
+        """
+        Send the vumi message as an HTTP request to the configured URL.
+        """
+        logger.debug("Consuming inbound message %s", message)
+        msg = junebug_from_msg(message)
+
+        headers = {}
+        if self.config.mo_message_url_auth_token is not None:
+            headers["Authorization"] = f"Token {self.config.mo_message_url_auth_token}"
+
+        # TODO: Handle timeouts
+        async with AsyncClient() as client:
+            resp = await client.post(
+                self.config.mo_message_url,
+                json=msg,
+                headers=headers,
+            )
+
+        if resp.status_code < 200 or resp.status_code >= 300:
+            logger.error(
+                LOG_MSG_HTTP_ERR,
+                {"code": resp.status_code, "body": resp.text, "message": msg},
+            )
+
+    async def handle_event(self, event: Event) -> None:
+        logger.debug("Consuming event %s", event)
+        event_hi = await self.state_cache.fetch_event_http_info(event.user_message_id)
+
+        if event_hi is None:
+            logger.warning(LOG_EV_URL_MISSING, {"event": event})
+            return
+
+        ev = junebug_from_ev(event, self.config.connector_name)
+
+        headers = {}
+        if event_hi.auth_token is not None:
+            headers["Authorization"] = f"Token {event_hi.auth_token}"
+
+        # TODO: Handle timeouts
+        async with AsyncClient() as client:
+            resp = await client.post(event_hi.url, json=ev, headers=headers)
+
+        if resp.status_code < 200 or resp.status_code >= 300:
+            logger.error(
+                LOG_EV_HTTP_ERR,
+                {"code": resp.status_code, "body": resp.text, "event": ev},
+            )
+
+        # TODO: Look up message_id and post event.
```

Comment:
Is this not done?

Reply:
Almost. It'll be done as soon as I remove the leftover TODO comments. 😛